### PR TITLE
Implementing first pass of veAllocate schema, handlers, and test cove…

### PR DESCRIPTION
**Pull request by idiom-bytes**
PR is WIP (not ready for merging)
- Tracer bullet for veAllocate
- Submitting this so we can track work + iterate towards a working subgraph

### DoD:
[x] - Designed gql schema
[x] Implemented event handlers
[x] Implemented getters & basic functionality
[ ] Implement tx,block,ts to create & update events
[ ] Implement & validate tests
[ ] Sign off on schema and handler design/implementation

### Identified issues outside current design scope:
See veAllocate.test.ts
[ ] veAllocate objects & interfaces inside of oceanprotocol/lib
[ ] implementing veAllocate + other functionality inside of contracts repo => `deploy_pool_template.js`. We need to add vyper dependencies + vyper compile steps into `deploy_docker.sh` and any other places required to deploy contracts via barge or other CI/CD workflows.

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -543,12 +543,51 @@ type NftUpdate @entity {
...
   id: ID!
   fixedRateTemplates: [String!]
   dispenserTemplates: [String!]
   ssTemplates: [String!]
 }
 
+# Not tracking allocationToId or idToAllocation
+# id = {user}
+type veAllocateUser @entity{
+  id: ID!
```
Comment (mihaisc) [inline schema.graphql:556]: minor suggestion, write the comment above the prop so the users see the comments in the schema like: 
```
"id = {user}"
id: ID!
```

If the comments were just provisional for internal use, ignore this :)

**Comment by alexcos20** [comment]
since ocean-contracts:dfve is not released as a npm package, for ve+df contracts we need to use artifacts exported by barge
This is fine for testing, will update once everything is done

**Comment by idiom-bytes** [comment]
Ok, the basic implementation for this is completed.
- dfpy veSetAllocation CHAINID amount exchangeId

ocean-subgraph:
- handles setAllocation + removeAllocation
- rolling totals for user, exchangeId, and each combination thereof works
- you can also see the Set/Remove events
- verified that tests that do not expect provider/aquarius are working
- verified that the data is looking correct

Queries
```
{
  veAllocations(first: 1000) {
    id
    allocationUser {
      id
    }
    allocationId{
      id
    }
    allocatedTotal
  }
}

{
  veAllocateUsers(first: 1000) {
    id
    allocatedTotal
  }
}

{
  veAllocateIds(first: 1000) {
    id
    allocatedTotal
  }
}

{
  veAllocationUpdates(first: 1000) {
    id
    veAllocation {
      id
      allocationUser{
        id
        allocatedTotal
      }
      allocationId {
        id
        allocatedTotal
      }
    }
    type
    allocatedTotal
  }
}
```

### Recommendation - Allocation Id:
The only thing I don't like about this, is that we need to do a bunch of offchain ingestion/calculations/mappings to understand which exchangeId + chainid the user is interacting with. It's not obvious by just looking at the data.

Consideration: Perhaps it might be a good idea if veAllocate keeps track of exchangeId + chainId, and can return that in the events. This way, we could use that information (exchangeId + chainId) in the subgraph. This would make the mapping a lot easier.

Screenshot from 2022-07-28 16-56-18

### Recommendation - veAllocation 
Add a cost to allocate. Users can currently DDOS the subgraph by injecting a bunch of garbage allocates. Make the fee from this go back to veOcean/farmers.

**Comment by idiom-bytes** [comment]
Fixed conflict.

New version of ocean-contract module has to be deployed so CI/CD can pass.

File: scripts/generatenetworkssubgraphs.js [hunk 2]
```diff
@@ -20,6 +20,14 @@ async function replaceContractAddresses() {
...
     console.log('Creating subgraph.yaml for ' + network)
     let subgraph = fs.readFileSync('./subgraph.template.yaml', 'utf8')
+    const subgraphVe = fs.readFileSync('./subgraph_ve.template.yaml', 'utf8')
+    if (addresses[network].veOCEAN) {
+      // fix identation , due to vs auto format (subgraph_ve.template is moved to left)
+      const lines = subgraphVe.split('\n')
+      for (let line = 0; line < lines.length; line++) {
+        subgraph += '  ' + lines[line] + '\n'
+      }
+    }
```
Comment (idiom-bytes) [inline scripts/generatenetworkssubgraphs.js:30]: Thanks Alex, this "looks good" although crazy to see the need to fix indentation from the build.

Comment (alexcos20) [reply scripts/generatenetworkssubgraphs.js:30]: My VS Code will "fix" indentation for every save on subgraph_ve.yaml, making impossible to join them :)

**Review by alexcos20 — APPROVED** [review]
I will merge it but:
 - DF contracts are not handled
 - there are no tests

**Pull request merged**